src/Driver/FlatFile/Driver.php

```diff
@@ -113,6 +117,21 @@ public function popMessage($queueName, $duration = 5)
         return [null, null];
     }
 
+    /**
+     * @param string $queueDir
+     * @param string $id
+     * @return array
+     */
+    private function process($queueDir, $id) {
+        $name = $queueDir.DIRECTORY_SEPARATOR.$id;
+        $newName = $name.'.proceed';
+
+        if (!@rename($name, $newName)) {
+            throw new InsufficientPermissionsException("Unable to process file: '{$name}'");
```

Comment:
I would prefer `throw new InsufficientPermissionsException('Unable to process file: '.$name);` for consistency